deeprank/features/atomicfeature_1.py

```diff
@@ -0,0 +1,703 @@
+                type_.append(self.patch_type[(altResName, at)])
+                vdw_data = self.vdw_param[self.patch_type[(altResName, at)]]
+                vdw_eps.append(vdw_data[0])
+                vdw_sigma.append(vdw_data[1])
+
+            elif (resName, at) in self.at_name_type_convertor:
+                type_.append(self.at_name_type_convertor[(resName, at)])
+                vdw_data = self.vdw_param[self.at_name_type_convertor[(
+                    resName, at)]]
+                vdw_eps.append(vdw_data[0])
+                vdw_sigma.append(vdw_data[1])
+
+            else:
+                type_.append('None')
+                vdw_eps.append(0.0)
+                vdw_sigma.append(0.0)
+                warnings.warn(f"Atom type {at} not found for "
+                              f"resType {resName} or patch type {altResName}. "
+                              f"Set vdw eps and sigma to 0.0.")
+
+        return vdw_eps, vdw_sigma, type_
+
+    def _get_charge(self, resName, altResName, atNames):
+        """Get the charge information.
+
+        Args:
+            resName (str): name of the residue
+            altResName (str): alternative name of the residue
+            atNames (list(str)): names of the atoms
+        """
+        # in case the resname is not valid
+        if resName not in self.valid_resnames:
+            q = [0.0] * len(atNames)
+            return q
+
+        # assign the charges
+        q = []
+        for at in atNames:
+            if (altResName, at) in self.patch_charge:
+                q.append(self.patch_charge[(altResName, at)])
+            elif (resName, at) in self.charge:
+                q.append(self.charge[(resName, at)])
+            else:
+                q.append(0.0)
+                warnings.warn(f"Atom type {at} not found for "
+                              f"resType {resName} or patch type {altResName}. "
+                              f"Set charge to 0.0.")
+        return q
+
+    ####################################################################
+    #
+    #   Simple charges
+    #
+    ####################################################################
+
+    def evaluate_charges(self, extend_contact_to_residue=False):
+        """Evaluate the charges.
+
+        Args:
+            extend_contact_to_residue (bool, optional): extend to res
+        """
+        if self.verbose:
+            print('-- Compute list charge for contact atoms only')
+
+        # extract information from the pdb2sq
+        xyz = np.array(self.sqldb.get('x,y,z'))
+        atinfo = self.sqldb.get(self.atom_key)
+
+        charge = np.array(self.sqldb.get('CHARGE'))
+
+        # define the dictionaries
+        charge_data = {}
+        charge_data_xyz = {}
+
+        # entire residue or not
+        if extend_contact_to_residue:
+            index_contact_atoms= self.extend_contact_to_residue()
+        else:
+            index_contact_atoms = self.contact_atom_ids
```

Comment:
I can't find a point in the code where the "self.contact_atom_ids" variable is set. Am I missing something?
This could potentially cause a runtime error.

Reply:
Noted